### PR TITLE
fix: fix gvg not exist

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

the gvg info will not be available when object is just uploaded, so when init the replicate task, it needs to first pick a GVG.

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### Potential Impacts
* add potential impacts for other components here
* ...

### CODE REVIEW

File: modular/manager/task_retry_scheduler.go [hunk 1]
```diff
@@ -202,7 +202,23 @@ func (s *TaskRetryScheduler) retryReplicateTask(meta *spdb.UploadObjectMeta) err
...
+		if err != nil || bucketInfo == nil {
+			log.Errorw("failed to get bucket by bucket name", "bucket", bucketInfo, "error", err)
+			return err
+		}
+		gvgMeta, err := s.manager.pickGlobalVirtualGroup(context.Background(), bucketInfo.BucketInfo.GlobalVirtualGroupFamilyId, storageParams)
+		log.Infow("pick global virtual group", "gvg_meta", gvgMeta, "error", err)
+		if err != nil {
+			return err
+		}
+		replicateTask.GlobalVirtualGroupId = gvgMeta.ID
```
Comment (will-2012) [inline modular/manager/task_retry_scheduler.go:217]: Maybe need `replicateTask.SecondaryEndpoints = gvgMeta.SecondaryEndpoints`.

File: modular/manager/task_retry_scheduler.go [hunk 2]
```diff
@@ -202,7 +202,23 @@ func (s *TaskRetryScheduler) retryReplicateTask(meta *spdb.UploadObjectMeta) err
 	replicateTask = &gfsptask.GfSpReplicatePieceTask{}
 	replicateTask.InitReplicatePieceTask(objectInfo, storageParams, s.manager.baseApp.TaskPriority(replicateTask),
 		s.manager.baseApp.TaskTimeout(replicateTask, objectInfo.GetPayloadSize()), s.manager.baseApp.TaskMaxRetry(replicateTask))
-	replicateTask.GlobalVirtualGroupId = meta.GlobalVirtualGroupID
+
+	if meta.GlobalVirtualGroupID == 0 {
```
Comment (will-2012) [inline modular/manager/task_retry_scheduler.go:206]: Add some comments to explain the scene.

Comment (alexgao001) [reply modular/manager/task_retry_scheduler.go:208]: added

File: modular/manager/task_retry_scheduler.go (hunk 1 above)
Comment (alexgao001) [reply modular/manager/task_retry_scheduler.go:219]: added

**Review by will-2012 — APPROVED** [review]

**Review by KeefeL — APPROVED** [review]

**Pull request merged**